pcdsdevices/positioner.py

```diff
@@ -0,0 +1,187 @@
+import time
+
+import numpy as np
+from ophyd.positioner import SoftPositioner
+from ophyd.signal import EpicsSignal, Signal
+from ophyd.utils import InvalidState
+
+from .interface import FltMvInterface
+from .utils import schedule_task
+
+
+class FuncPositioner(FltMvInterface, SoftPositioner):
+    """
+    Class for hacking together a positioner-like object.
+
+    Before you use this, make sure your use case cannot be easily handled by a
+    `PVPositioner` or by a `PseudoPositioner`.
+
+    This should be fast to set up, but has the following limitations:
+    - Inspection tools will not work properly, in general
+    - typhos, happi, and similar tools will not work
+    - It is not possible to know ahead of time if any control layer signals
+      poked by this device are connected.
+    - Metadata about the control layer will not be provided.
+    - It's not possible to intelligently subscribe to control layer signals.
+      Everything will be periodic polls on background threads.
+    - Session performance may be negatively impacted by slow polling functions.
+
+    Parameters
+    ----------
+    name : str, keyword-only
+        The name to use when we refer to this positioner.
+
+    move : func, keyword-only
+        The function to call to cause the device to move.
+        The signature must be def fn(position)
```

Comment:
I'd recommend to check signatures - something like what I had just pushed into caproto - to make the lives of users easier:
https://github.com/caproto/caproto/pull/736/commits/ae59914652226d0f2d8525377c586e40ea41e035

Reply:
Good call, I'll take a closer look at this example and implement something similar

Reply:
Decided to keep it fairly simple: https://github.com/pcdshub/pcdsdevices/pull/786/commits/b5aee231e5040d247aaa633059b8478e9763d54b

Reply:
Nice 👍